SimDataFormats/CaloAnalysis/interface/MtdSimLayerCluster.h

```diff
@@ -0,0 +1,300 @@
+// Author: Aurora Perego, Fabio Cossutti - aurora.perego@cern.ch, fabio.cossutti@ts.infn.it
+// Date: 05/2023
+
+#ifndef SimDataFormats_CaloAnalysis_MtdSimLayerCluster_h
+#define SimDataFormats_CaloAnalysis_MtdSimLayerCluster_h
+
+#include "DataFormats/HepMCCandidate/interface/GenParticle.h"
+#include "DataFormats/Math/interface/LorentzVector.h"
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "SimDataFormats/Track/interface/SimTrack.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include <vector>
+
+//
+// Forward declarations
+//
+class SimTrack;
+class EncodedEventId;
+
+/** @brief Monte Carlo truth information used for tracking validation.
+ *
+ * Object with references to the original SimTrack and parent and daughter
+ * TrackingVertices. Simulation with high (~100) pileup was taking too much
+ * memory so the class was slimmed down and copies of the SimHits were removed.
+ *
+ * @author original author unknown, re-engineering and slimming by Subir Sarkar
+ * (subir.sarkar@cern.ch), some tweaking and documentation by Mark Grimes
+ * (mark.grimes@bristol.ac.uk).
+ * @date original date unknown, re-engineering Jan-May 2013
+ */
+class MtdSimLayerCluster {
+  friend std::ostream &operator<<(std::ostream &s, MtdSimLayerCluster const &tp);
```

Comment:
@AuroraPerego which is the idea behind making this class fully independent on ```SimCluster```, while sharing most of its structure with it?